### PR TITLE
feat: deposit collection for interchain queries

### PR DESCRIPTION
**Pull request by albertandrejev**
This PR will add documentation for deposit collection for interchain queries.

Related PR: https://github.com/neutron-org/neutron/pull/48

Part of the https://p2pvalidator.atlassian.net/browse/NTRN-184

### CODE REVIEW

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
I think there should be a bit more info about deposit mechanism, maybe even a separate subparagraph: how deposit works, how user can return it, why and when deposit amount can change (gov params), etc.
I mean, all that info already here, but it's hidden in some `State` paragraph, and I think it should be much visible, cause it's out core mechanism.

Also I think [ICQ tutorial](https://github.com/neutron-org/neutron-docs/blob/main/docs/tutorials/cosmwasm_icq.md) should be also updated. At least  some small note/link to the `Deposit` paragraph should be there

**Comment by albertandrejev** [comment]
I will put it into [overview.md](https://github.com/neutron-org/neutron-docs/blob/main/docs/neutron-core/interchain-queries/overview.md)

File: docs/neutron-core/interchain-queries/overview.md [hunk 1]
```diff
@@ -41,4 +41,11 @@ If your contract does not have such checks, malicious relayer can send a fully v
...
 
-You can see more info, examples and recommendations about proper transactions result handling [here](https://github.com/neutron-org/neutron-contracts/blob/main/contracts/neutron_interchain_txs/src/contract.rs#L335).
\ No newline at end of file
+You can see more info, examples and recommendations about proper transactions result handling [here](https://github.com/neutron-org/neutron-contracts/blob/main/contracts/neutron_interchain_txs/src/contract.rs#L335).
+
+## Query creation deposit
+In order to clean up ledger from not used, outdated queries special deposit mechanism is used. [RegisteredQuery](https://github.com/neutron-org/neutron/blob/4313d35f8082dc124c5fe9491870720bbd3a5052/proto/interchainqueries/genesis.proto#L9 contains `deposit` field, this field is used to collect escrow payment for query creation. In order to return escrow payment owner or anybode else should issue `RemoveInterchainQuery` message.
```
Comment (sotnikov-s) [inline docs/neutron-core/interchain-queries/overview.md:47]: broken link to `RegisteredQuery`: missing trailing `)`

Comment (sotnikov-s) [reply docs/neutron-core/interchain-queries/overview.md:47]: typo: In order to return escrow payment owner or anybod**e** else

Comment (sotnikov-s) [reply docs/neutron-core/interchain-queries/overview.md:47]: > In order to return escrow payment owner or anybode else should issue `RemoveInterchainQuery` message. 

the core conclusion from this sentence seems to be like `omg anybody can take my money, wat?`. I'd move the whole `RemoveInterchainQuery` clarification to the next paragraph, and here just left:
> In order to return escrow payment, a `RemoveInterchainQuery` message should be issues.

Comment (sotnikov-s) [reply docs/neutron-core/interchain-queries/overview.md:47]: I'd also add a user-friendly description after this terms-math-tech one. e.g.:

In other words, it is expected of the query owner to remove its queries when they are not needed anymore. If a query hasn't been in use for the `query_submit_timeout` and owner hasn't removed it, network users are granted with an opportunity to clean the chain up and raise assets for it.

**Review by sotnikov-s — CHANGES_REQUESTED** [review]

**Pull request merged**